src/id/id.c

```diff
@@ -74,7 +74,7 @@ static inline int _fit_width(uint32_t val, unsigned width) {
   uint32_t mask;
   uint32_t b;
 
-  while (offset >= width) {
+  while (offset > width) {
```

Comment:
What's the expected behavior when width == 32? Even though a warning is generated by older gcc, I can see the behavior that it will rotate it and `1<<32` becomes 1 again. If that's the case, then mask will become 0 and this function will return 32 in that case.
Do we want to return 32 if width was sent in as 32?

Reply:
The mask is 0x1000 0000 for width 1, 0x1100 0000 for width 2 so on till 0xFFFF FFFE for width 31. So I think for width 32 mask should be 0xFFFF FFFF.  To avoid, compiler dependency, I will add a check for width = 32.

Reply:
Sounds good. I guess then the while condition shouldn't be changed ?

Reply:
Oh mistakenly, I changed the condition also
will raise new PR

Reply:
Now everything looks fine